### ISSUE TITLE
pscustomobject in hastables are ommited in conversion (v0.4.8 - v0.4.10)

### ISSUE THREAD

**Issue by f-bader** [cloudbase/powershell-yaml#177]
Since version 0.4.8 up until 0.4.10 a pscustomobject added to a hashtable is ommited from the conversion

```
$OrderedHashtable = [ordered]@{}
$OrderedHashtable.Add("TestKey",[pscustomobject]@{"aggregationKind"="SingleAlert"})
$OrderedHashtable | ConvertTo-Yaml
```
### v0.4.7
![image](https://github.com/user-attachments/assets/706a16a8-0fda-4484-b260-87ddf6540330)

### v0.4.10
![image](https://github.com/user-attachments/assets/2c9714ef-e6bc-4d38-a4d9-14aaa90d83d0)

### Current workaround

![image](https://github.com/user-attachments/assets/0ee490c7-2d3f-4e80-8bcc-e4cc9bd65cb1)

**Comment by gabriel-samfira** [comment]
Would you mind doing an `Update-Module powershell-yaml` and see if it still happens?

**Comment by gabriel-samfira** [comment]
```powershell
PS /home/gabriel> (Get-Module powershell-yaml).version.ToString()
0.4.11
PS /home/gabriel> import-module powershell-yaml       
PS /home/gabriel> $OrderedHashtable = [ordered]@{}
PS /home/gabriel> $OrderedHashtable.Add("TestKey",[pscustomobject]@{"aggregationKind"="SingleAlert"})
PS /home/gabriel> $OrderedHashtable | ConvertTo-Yaml
TestKey:
  aggregationKind: SingleAlert
```

**Comment by f-bader** [comment]
![image](https://github.com/user-attachments/assets/64aa11f7-6108-4b21-a4c7-b9e159e8514a)

Looks fixed in 0.4.11 - Thank you very much

**Comment by gabriel-samfira** [comment]
Added a bunch of tests as well (will add more after I free myself up). I made a mess this December. The last 3 releases fixed the mess I did 4 releases ago. Hopefully things work well in the latest version and we can have a long while without any updates.

Thanks for your patience while I sorted out my mess!
